synchronization/semaphore.cc

```diff
@@ -48,8 +48,48 @@ class PlatformSemaphore {
 
 }  // namespace flutter
 
-#else  // OS_MACOSX
+#elif OS_WIN
+#include <windows.h>
+
+namespace flutter {
+
+class PlatformSemaphore {
+ public:
+  explicit PlatformSemaphore(uint32_t count)
+      : _sem(CreateSemaphore(NULL, count, 1024, NULL)) {}
```

Comment:
Out of curiosity, why a max count of 1024?

Reply:
That's a good question... I'd like to ask July 2017 Ben the same thing. Either way, we need to provide a maximum count and there's no limit (and 1024 seems to work for now...).

Reply:
Looks like it's a long -- maybe just `LONG_MAX`?

Reply:
Makes sense. Done!